lib/mix/lib/mix/tasks/test.ex

```diff
@@ -63,6 +65,14 @@ defmodule Mix.Tasks.Test do
       end
     end
 
+    # protocol definitions have no testable code they shouldn't show up in coverage results
+    # see #8825
+    defp discard_protocol_definitions(coverage_results) do
+      Enum.reject(coverage_results, fn {{module, _}, _} ->
+        function_exported?(module, :__protocol__, 1)
```

Comment:
Not sure if this check suffices here of it should rather be like `Module.protocol?/1`:

```
  defp protocol?(module) do
    Code.ensure_loaded?(module) and function_exported?(module, :__protocol__, 1) and
      module.__protocol__(:module) == module
  end
```

Speaking of which, as other parts of the code base also check for `function_exported?(module, :__protocol__)` (info.ex and xref.ex) it might be worth considering (from my point of views) to make `Module.protocol?/1` public - would have also kind of hoped to find it in a public function somewhere :)